src/state_machine/signer/mod.rs

```diff
@@ -0,0 +1,578 @@
+    /// A bad state change was made
+    #[error("Bad State Change: {0}")]
+    BadStateChange(String),
+}
+
+/// A state machine for a signing round
+pub struct SigningRound<Signer: SignerTrait> {
+    /// current DKG round ID
+    pub dkg_id: u64,
+    /// current signing round ID
+    pub sign_id: u64,
+    /// current signing iteration ID
+    pub sign_iter_id: u64,
+    /// the threshold of the keys needed for a valid signature
+    pub threshold: u32,
+    /// the total number of signers
+    pub total_signers: u32,
+    /// the total number of keys
+    pub total_keys: u32,
+    /// the Signer object
+    pub signer: Signer,
+    /// the Signer ID
+    pub signer_id: u32,
+    /// the current state
+    pub state: State,
+    /// map of party_id to the polynomial commitment for that party
+    pub commitments: BTreeMap<u32, PolyCommitment>,
+    /// map of decrypted DKG private shares
+    pub decrypted_shares: HashMap<u32, HashMap<u32, Scalar>>,
+    /// invalid private shares
+    pub invalid_private_shares: Vec<u32>,
+    /// public nonces for this signing round
+    pub public_nonces: Vec<PublicNonce>,
+    /// the private key used to sign messages sent over the network
+    pub network_private_key: Scalar,
+    /// the public keys for all signers and coordinator
+    pub public_keys: PublicKeys,
+}
+
+impl<Signer: SignerTrait> SigningRound<Signer> {
+    /// create a SigningRound
+    pub fn new(
+        threshold: u32,
+        total_signers: u32,
+        total_keys: u32,
+        signer_id: u32,
+        key_ids: Vec<u32>,
+        network_private_key: Scalar,
+        public_keys: PublicKeys,
+    ) -> Self {
+        assert!(threshold <= total_keys);
+        let mut rng = OsRng;
+        let signer = Signer::new(
+            signer_id,
+            &key_ids,
+            total_signers,
+            total_keys,
+            threshold,
+            &mut rng,
+        );
+        debug!(
+            "new SigningRound for signer_id {} with key_ids {:?}",
+            signer_id, &key_ids
+        );
+        SigningRound {
+            dkg_id: 0,
+            sign_id: 1,
+            sign_iter_id: 1,
+            threshold,
+            total_signers,
+            total_keys,
+            signer,
+            signer_id,
+            state: State::Idle,
+            commitments: BTreeMap::new(),
+            decrypted_shares: HashMap::new(),
+            invalid_private_shares: Vec::new(),
+            public_nonces: vec![],
+            network_private_key,
+            public_keys,
+        }
+    }
+
+    fn reset<T: RngCore + CryptoRng>(&mut self, dkg_id: u64, rng: &mut T) {
+        self.dkg_id = dkg_id;
+        self.commitments.clear();
+        self.decrypted_shares.clear();
+        self.invalid_private_shares.clear();
+        self.public_nonces.clear();
+        self.signer.reset_polys(rng);
+    }
+
+    ///
+    pub fn process_inbound_messages(
+        &mut self,
+        messages: Vec<Packet>,
```

Comment:
Similarly this is not required. It could be messages: &[Packet] which will save us clones outside of this module.